bitcoin/src/blockdata/script/borrowed.rs

```diff
@@ -133,6 +133,11 @@ impl Script {
         let inner = unsafe { Box::from_raw(rw) };
         ScriptBuf(Vec::from(inner))
     }
+
+    /// Writes the human-readable assembly representation of the script to the formatter.
+    pub fn fmt_asm(&self, f: &mut dyn fmt::Write) -> fmt::Result {
+        bytes_to_asm_fmt(self.as_ref(), f)
+    }
```

Comment:
Thinking about this again, this does the same thing `Display` does, so we might stay more conservative and not move this. Implement the one in extension trait by simply calling `fmt::Display::fmt(self, f)`

Reply:
Not sure what to make of this comment now, `Display` calls `fmt_asm` so it has to move, right?

Reply:
`Display` can just call `bytes_to_asm_fmt` and then the not-moved method can use `write!`. Honestly, I think the method is pretty much useless and everyone is using `Display` instead.

Reply:
Ok, after much ado, we now deprecate `fmt_asm` and `to_asm_string`. We still use them in fuzzing. Doing all this made me create a `primitives` fuzz target directory and move the `script_bytes_to_asm_fmt` fuzz test over there - phew!